code/game/objects/structures/ai_core.dm

```diff
@@ -253,28 +253,34 @@ That prevents a few funky behaviors.
 //The type of interaction, the player performing the operation, the AI itself, and the card object, if any.
 
 
-/atom/proc/transfer_ai(interaction, mob/user, mob/living/silicon/ai/AI, obj/item/device/aicard/card)
-	if(istype(card))
-		if(card.flush)
-			to_chat(user, "<span class='boldannounce'>ERROR</span>: AI flush is in progress, cannot execute transfer protocol.")
-			return 0
-	return 1
+/atom/proc/transfer_ai(interaction, mob/user, mob/living/silicon/ai/AI, obj/card)
+	var/datum/component/ai_storage/C = card.GetComponent(/datum/component/ai_storage)
+	if(!C)
+		return
+	if(C.wiping)
+		to_chat(user, "<span class='boldannounce'>ERROR</span>: AI flush is in progress, cannot execute transfer protocol.")
+		return FALSE
+	return TRUE
 
 
-/obj/structure/AIcore/transfer_ai(interaction, mob/user, mob/living/silicon/ai/AI, obj/item/device/aicard/card)
+/obj/structure/AIcore/transfer_ai(interaction, mob/user, mob/living/silicon/ai/AI, obj/card)

```

Comment:
The core itself should have a storage component which does the transform with a callback